unmarshaler.go

```diff
@@ -11,7 +11,7 @@ type MultiUnmarshaler[H header.Header[H]] struct {
 }
 
 func (d MultiUnmarshaler[H]) List() []ProofType {
-	types := make([]ProofType, len(d.Unmarshalers))
+	types := make([]ProofType, 0, len(d.Unmarshalers))
 	for tp := range d.Unmarshalers {
 		types = append(types, tp)
 	}
```

Comment:
It looks fine as is. Looks like it prealocs with capacity and appends later, so allocated space will be used. If allocated with size, it will contain bunch of nil unmarshallers

Reply:
Hm, I'm not sure I get it.

`List` method returns all the known `ProofType` (at least how method declarations looks to me). However, we will return `x` empty strings (`ProofType` is a string) + `x` proof types from `.Unmarshalers` field.

See link in the PR description.

Reply:
You can add `t.Logf("len: %d vs %d", len(unmarshaler.List()), len(unmarshaler.Unmarshalers))` in `fraudserv/service_test.go` (before `t.Cleanup`)

Reply:
My brain somehow messed up what was added with what was removed removed.